components/organisms/InsightPage/DeleteInsightPageModal.tsx

```diff
@@ -0,0 +1,68 @@
+import { FC, useState } from "react";
+import Button from "components/atoms/Button/button";
+import TextInput from "components/atoms/TextInput/text-input";
+import Text from "components/atoms/Typography/text";
+import Title from "components/atoms/Typography/title";
+import { Dialog, DialogContent, DialogTitle } from "components/molecules/Dialog/dialog";
+
+interface ModalProps {
+  open: boolean;
+  submitted: boolean;
+  pageName: string;
+  onClose: () => void;
+  onConfirm: () => Promise<void>;
+}
+
+const DeleteInsightPageModal:FC<ModalProps> = ({
+  open = false,
+  submitted = false,
+  pageName,
+  onConfirm,
+  onClose
+}) => {
+
+  const [input, setInput] = useState("");
+
+  const handleOnNameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setInput(e.target.value);
+  };
+
+  const handleOnConfirm = async () => {
+    if (input !== "DELETE") return;
+    await onConfirm();
+  };
+
+  const handleOnClose = async () => {
+    onClose();
+  };
+
+  return (
+    <Dialog
+      open={open}
+    >
+      <DialogContent>
+        <DialogTitle>
+          <Title level={3}>Delete Page</Title>
+        </DialogTitle> 
+
+        <Text>Are you sure you want to delete  <span className="font-bold text-light-slate-12">{`${pageName}`}</span>?</Text>
+        <Text>If you have data on this page that your team is using it would be difficult for your team to get access to track your project.</Text>
+        <Text> <span className="font-bold text-light-slate-12">This action cannot be undone</span></Text>
+        <Text>Type DELETE in all caps to confirm</Text>
+
+        <TextInput onChange={handleOnNameChange} value={input} />
+
+        <div className="flex gap-3">
+          <Button disabled={submitted} onClick={handleOnConfirm} variant="default" className="bg-light-red-6 border border-light-red-8 hover:bg-light-red-7 text-light-red-10">
```

Comment:
This button needs to be disabled when the input hasn't matched `DELETE` and also show a loading indicator when the delete call is in progress

Reply:
When the delete request call is in progress the button gets visibly disabled. Isn't it enough because the delete request takes very little time? If not where should I place the loading indicator?